### PR TITLE
[core][dashboard] Change py-spy native default to false

### LINKED ISSUE #30566: [Dashboard] Getting stacktrace from the dashboard triggers "failed to get os threadid" when `--native` option is used
### What happened + What you expected to happen

https://console.anyscale.com/o/anyscale-internal/projects/prj_FKRmeV5pA6X72aVscFALNC32/clusters/ses_GrgDrjtJWSw9A6wM9sbSwfwe?command-history-section=command_history

When I click the `stack trace` for the JobSupervisorActor, I saw the following error.

```
Failed to execute `sudo -n $(which py-spy) dump -p 830 --native`.

Note that this command requires `py-spy` to be installed with root permissions. You
can install `py-spy` and give it root permissions as follows:
  $ pip install py-spy
  $ sudo chown root:root `which py-spy`
  $ sudo chmod u+s `which py-spy`

Alternatively, you can start Ray with passwordless sudo / root permissions.

=== stdout ===
Process 830: ray::JobSupervisor
Python v3.7.7 (/home/ray/anaconda3/bin/python3.7)



=== stderr ===
Error: failed to get os threadid
```

Even when I run the same command from the terminal, it failed with the same error

Based on the search, I found only one issue, https://github.com/benfred/py-spy/issues/490, but this seems irrelevant to our case (it says it should work if this command runs inside a container). 

We may need to figure out the root cause or remove --native as a default option (or we can allow users to turn on/off the flag). 

### Versions / Dependencies

master

### Reproduction script

not sure how to reproduce, but it is easy to find this problem from nightly test.

### Issue Severity

_No response_

### PR DESCRIPTION
**Pull request by rickyyx**
Signed-off-by: rickyyx <rickyx@anyscale.com>

## Related issue number
Closes https://github.com/ray-project/ray/issues/30566

## Checks

Run agent test which failed to get profiling/stacktrace now runs OK: 
image

Changing query to `native=1` will fail the query again:
image

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — COMMENTED** [review]
Can we add native=0 to the href as well in the frontend? This is to surface that you can set native=1 to turn it on.

**Review by ericl — COMMENTED** [review]
I think you'll also need to add/fix the test in test_dashboard_profiler.py, which assumes native frames right now.

**Comment by rickyyx** [comment]
> I think you'll also need to add/fix the test in test_dashboard_profiler.py, which assumes native frames right now.

Oh I see, had some troubles running the test locally. Will take a look at the CI run.

**Review by rkooo567 — COMMENTED** [review]
Is it possible to test the change?

**Comment by ericl** [comment]
Yes, we should add a test case to the profiler suite to test the flag.

**Comment by rickyyx** [comment]
Modified the test to test native vs non-native profiling. 

Wondering if there's any good way for us to let users know the default is non-native, and they could change the parameter for native profiling? 

Have we thought about default to native, and retry with non native if the native profiling fails?

**Review by ericl — COMMENTED** [review]
It's not a bad idea. Like with `native=1`, try-catch native and fallback to non-native? Then we could enable native by default.

Not sure if there are any downsides here.

**Review by ericl — APPROVED** [review]

File: python/ray/tests/test_dashboard_profiler.py [hunk 1]
```diff
@@ -65,15 +67,18 @@ def get_actor_stack():
...
         response.raise_for_status()
         assert response.headers["Content-Type"] == "image/svg+xml", response.headers
         content = response.content.decode("utf-8")
         print(content)
         # Sanity check we got the flame graph SVG.
         assert "<!DOCTYPE svg" in content, content
         assert "do_stuff_infinite" in content, content
-        assert "ray::core" in content, content
+        if native == "1":
+            assert "ray::core" in content, content
```
Comment (ericl) [inline python/ray/tests/test_dashboard_profiler.py:81]: else assert this doesn't exist.

**Comment by rickyyx** [comment]
> Like with `native=1`, try-catch native and fallback to non-native? Then we could enable native by default.

Right - because it seems it's only with some specific corner cases that native profiling fails (not sure how frequent this happens, spot checked around 15 release tests clusters, and I think around 30% of them not able to get native profiling). 

We could still have this option in the url, but native will be on by default.

**Comment by ericl** [comment]
> Right - we could probably hide the option from url? If native=1 in url, but then we not showing the native due to fallback, it might be a bit confusing?

Maybe let's do this in a separate PR then, to keep the cherry pick small / uncontroversial.

**Pull request merged**